core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -148,6 +149,7 @@
 
     // must initialized in doStart(), when we have the allocationService set
     private volatile NodeJoinController nodeJoinController;
+    private volatile NodeRemovalClusterStateTaskExecutor nodeRemovalExecutor;
```

Comment:
It can't be final, it's constructed on zen discovery start because the elect master service isn't available until then.